Wikipedia/Code/ExploreViewController.swift

```diff
@@ -1105,6 +1106,65 @@ extension ExploreViewController {
             yirDataController.hasPresentedYiRFeatureAnnouncementModel = true
         }
     }
+    
+    private func shouldShowSearchWidgetAnnouncement() -> Bool {
+        // Check if user has already seen the announcement
+        if UserDefaults.standard.wmf_didShowSearchWidgetFeatureAnnouncement {
+            return false
+        }
+        
+        // Check if current date is before the temporary date (September 30, 2025)
+        let calendar = Calendar.current
+        var expiryDateComponents = DateComponents()
+        expiryDateComponents.year = 2025
+        expiryDateComponents.month = 9
+        expiryDateComponents.day = 30
+        
+        guard let expiryDate = calendar.date(from: expiryDateComponents) else {
+            return false
+        }
+        
+        let currentDate = Date()
+        return currentDate <= expiryDate
+    }
+    
+    private func markSearchWidgetAnnouncementAsSeen() {
+        UserDefaults.standard.wmf_didShowSearchWidgetFeatureAnnouncement = true
+    }
+    
+    private func presentSearchWidgetAnnouncement() {
+        // Check if the announcement should show
+        guard shouldShowSearchWidgetAnnouncement() else {
+            return
+        }
+        
+        let title = CommonStrings.searchWidgetAnnouncementTitle
+        let body = CommonStrings.searchWidgetAnnouncementBody
+        let primaryButtonTitle = CommonStrings.gotItButtonTitle
+        
+        let backgroundImage = UIImage(named: "widgetgroup")
+        
+        let viewModel = WMFFeatureAnnouncementViewModel(title: title,body: body,
+ primaryButtonTitle: primaryButtonTitle, image: nil, backgroundImage: backgroundImage,
+            gifName: nil, altText: CommonStrings.searchWidgetAnnouncementBody,
+            primaryButtonAction: { [weak self] in
+                self?.dismiss(animated: true)
+            },
+            closeButtonAction: { [weak self] in
+                self?.dismiss(animated: true)
+            }
+        )
+        
+         if let searchBarButtonItem = navigationItem.leftBarButtonItem {
+            announceFeature(viewModel: viewModel, sourceView: nil, sourceRect: nil, barButtonItem: searchBarButtonItem)
+            // Mark as seen after successful presentation
+            markSearchWidgetAnnouncementAsSeen()
+        } else if let profileBarButtonItem = navigationItem.rightBarButtonItem {
+            announceFeature(viewModel: viewModel, sourceView: nil, sourceRect: nil, barButtonItem: profileBarButtonItem)
+            // Mark as seen after successful presentation
+            markSearchWidgetAnnouncementAsSeen()
+        }
```

Comment:
We don't have a search bar button so that check will always fail.

Here's an optional follow-up if you need a challenge:
The presentation of the feature doesn't need to be tied to the profile button. It was done for YiR because the entry point for that feature was in the profile view. Adding this pointing to the profile view (it only shows on iPad, on iPhone is the modal sheet) could lead the user to look for something there.

You could look into the method `announceFeature(...` in `WMFFeatureAnnouncing`. Right now we check if there's a source rect or a button. 
You could create a default presentation way in case there's neither a source view nor a button, and have it display centralized on the iPad screen.